runtime/integration-tests/src/generic/cases/block_rewards.rs

```diff
@@ -0,0 +1,202 @@
+	instances::BlockRewards,
+};
+use sp_runtime::traits::{Get, Zero};
+
+use crate::{
+	generic::{
+		config::Runtime,
+		env::Env,
+		envs::runtime_env::RuntimeEnv,
+		utils,
+		utils::{
+			currency::cfg,
+			genesis::{self, Genesis},
+		},
+	},
+	utils::accounts::Keyring,
+};
+
+crate::test_for_runtimes!(all, block_rewards_api);
+fn block_rewards_api<T: Runtime>() {
+	const STAKER: Keyring = Keyring::Alice;
+
+	RuntimeEnv::<T>::default().parachain_state_mut(|| {
+		let group_id = 1u32;
+		let amount = 100 * CFG;
+
+		utils::give_balance::<T>(STAKER.id(), T::ExistentialDeposit::get() + amount);
+
+		assert_ok!(pallet_rewards::Pallet::<T, BlockRewards>::attach_currency(
+			CurrencyId::Native,
+			group_id,
+		));
+
+		assert_ok!(pallet_rewards::Pallet::<T, BlockRewards>::deposit_stake(
+			CurrencyId::Native,
+			&STAKER.id(),
+			amount,
+		));
+
+		assert_ok!(
+			pallet_rewards::Pallet::<T, BlockRewards>::distribute_reward(200 * CFG, [group_id])
+		);
+
+		assert_eq!(
+			T::Api::list_currencies(RewardDomain::Block, STAKER.id()),
+			vec![CurrencyId::Native]
+		);
+
+		assert_eq!(
+			T::Api::compute_reward(RewardDomain::Block, CurrencyId::Native, STAKER.id()),
+			Some(200 * CFG)
+		)
+	});
+}
+
+crate::test_for_runtimes!([development], collator_list_synchronized);
+fn collator_list_synchronized<T: Runtime>() {
+	RuntimeEnv::<T>::from_parachain_storage(
+		Genesis::default()
+			.add(genesis::balances::<T>(
+				T::ExistentialDeposit::get() + cfg(100),
+			))
+			.add(genesis::invulnerables::<T>(vec![Keyring::Admin]))
+			.add(genesis::session_keys::<T>())
+			.add(genesis::block_rewards::<T>(vec![Keyring::Admin]))
+			.storage(),
+	)
+	.parachain_state_mut(|| {
+		let collators_1 = vec![Keyring::Alice.id(), Keyring::Bob.id()];
+		let collators_2 = vec![
+			Keyring::Charlie.id(),
+			Keyring::Dave.id(),
+			Keyring::Eve.id(),
+			Keyring::Ferdie.id(),
+		];
+
+		// SESSION 0 -> 1;
+		apply_and_check_session::<T>(1, collators_1.clone(), vec![]);
+
+		// SESSION 1 -> 2;
+		apply_and_check_session::<T>(3, collators_2.clone(), vec![]);
+
+		// SESSION 2 -> 3;
+		apply_and_check_session::<T>(7, vec![], vec![Keyring::Alice.id()]);
+
+		assert!(collators_1.iter().all(|c| has_reward::<T>(c)));
+		assert!(collators_2.iter().all(|c| !has_reward::<T>(c)));
+
+		// SESSION 3 -> 4;
+		apply_and_check_session::<T>(6, vec![], vec![]);
+
+		assert!(collators_2.iter().all(|c| has_reward::<T>(c)));
+	});
+}
+
+fn apply_and_check_session<T: Runtime>(
+	collator_count: u32,
+	joining: Vec<AccountId>,
+	leaving: Vec<AccountId>,
+) {
+	for collator in &joining {
+		pallet_collator_selection::Pallet::<T>::register_as_candidate(
+			RawOrigin::Signed(collator.clone()).into(),
+		)
+		.unwrap();
+	}
+
+	for collator in &leaving {
+		pallet_collator_selection::Pallet::<T>::leave_intent(
+			RawOrigin::Signed(collator.clone()).into(),
+		)
+		.unwrap();
+	}
+
+	frame_system::Pallet::<T>::reset_events();
+
+	pallet_session::Pallet::<T>::rotate_session();
```

Comment:
BTW, I've played with this trying to substitute `pass_n(Period)` to `rotate_session()`, which maps to the same behavior and allow us getting rid of `fast-runtime` feature in integration-test and simplify a bit the test.

If you have some thoughts against it @wischli , I can revert this.